examples/healthData/src/translator.ts

```diff
@@ -0,0 +1,75 @@
+import {Result, TypeChatLanguageModel, createJsonTranslator, TypeChatJsonTranslator} from "typechat";
+
+type ChatMessage = {
+    source: "system" | "user" | "assistant";
+    body: string;
+};
+
+export interface TranslatorWithHistory<T extends object> {
+    _chatHistory: ChatMessage[];
+    _maxPromptLength: number;
+    _additionalAgentInstructions: string;
+    _translator: TypeChatJsonTranslator<T>;
+    translate(request: string): Promise<Result<T>>;
+}
+
+export function createHealthDataTranslator<T extends object>(model: TypeChatLanguageModel, schema: string, typename: string, additionalAgentInstructions: string): TranslatorWithHistory<T> {
+    const _chatHistory: ChatMessage[] = [];
+    const _maxPromptLength = 2048;
+    const _additionalAgentInstructions = additionalAgentInstructions;
+    
+    const _translator = createJsonTranslator<T>(model, schema, typename);
+    _translator.createRequestPrompt = _create_request_prompt;
+    
+    const customtranslator: TranslatorWithHistory<T> = {
+        _chatHistory,
+        _maxPromptLength,
+        _additionalAgentInstructions,
+        _translator,
+        translate,
+    };
+
+    return customtranslator;
+
+    async function translate(request: string): Promise<Result<T>> {
+        const response = await _translator.translate(request);
+        if (response.success) {
+            _chatHistory.push({ source: 'assistant', body: JSON.stringify(response.data) });
```

Comment:
You probably didn't mean to stringify the body, since you'll end up stringifying the string version when you generate `historyStr`.

```suggestion
            _chatHistory.push({ source: "assistant", body: response.data });
```